### PR TITLE
fix: userLookup API should return all users [DHIS2-11317]

### PR DESCRIPTION
**Pull request by jbee**
### Summary
PR makes sure a user with the same role as other users can see those users when doing a search in the `/userLookup?query` API.

The query already had an override flag to bypass the _same role_ check.

### Automatic Testing
I added a controller test that setups a test scenario where all users including the user doing the lookup have a common role.
Then the user does a search for one of the users and finds the user. 

When the added `.setCanSeeOwnUserAuthorityGroups( true )` is removed (or set to `false`) this tests fails as no user is found.

### Manual Testing
Pretty much what the tests do. 

1. Create a role and add both the user doing the lookup as well as the user that should be found to it. 
2. switch to the lookup user
3. do the lookup `/userLookup?query=<name>` and check the other user with the same role is found (as well as other users that match the search term obviously)

### CODE REVIEW

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/user/UserLookupController.java
Comment (Bekkalizer) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/user/UserLookupController.java:88]: I see that the current method name is "canSeeOwnUserAuthorityGroups" - but the authority is CAN_GRANT_OWN_USER_AUTHORITY_GROUPS. I realize that this name was inherited from the previous behavior, but it seems that we could do a little cleanup here. All users should be able to "**see**" their own groups and autos - but they should not be able to "**grant**" it.
This is not a big deal, and I'm not sure how much work it would be to rewrite this, I'll let you decide wether to implement a change to the naming and logic, or wether to keep it like this.

Comment (Bekkalizer) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/user/UserLookupController.java:88]: I would rather have it included before hard freeze tonight than have it perfect :)

Comment (jbee) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/user/UserLookupController.java:88]: @larshelge I think you added the logic in question. Maybe you can help us understand why it is needed or if this was a misunderstanding and always just should have targeted just the granting.

But yes, this is a different change we can and should make in another PR - also to get this in 2.37?

**Review by netroms — APPROVED** [review]

**Review by stian-sandvold — APPROVED** [review]

**Comment by stian-sandvold** [comment]
Approved, pending @Bekkalizer's comment :)

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/user/UserLookupController.java
Comment (larshelge) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/user/UserLookupController.java:88]: I think this is in fact correct, as this method is just fetching/reading user information. We have other endpoints for writing/modifying/granting access.

**Pull request merged**
